pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1520,7 +1536,10 @@ public void seek(long timestamp) throws PulsarClientException {
         cnx.sendRequestWithId(seek, requestId).thenRun(() -> {
             log.info("[{}][{}] Successfully reset subscription to message id {}", topic, subscription, messageId);
             acknowledgmentsGroupingTracker.flushAndClean();
-            lastDequeuedMessage = messageId;
+
+            seekMessageId = new BatchMessageIdImpl((MessageIdImpl) messageId);
+            duringSeek.set(true);
```

Comment:
Hi! Looks like this commit is a breaking change, it broke my unit test. The logic of unit test is as follows
1) Read 10 messages
2) Seek back to the first
3) Read 10 messages again
Earlier `hasMessageAvailableAsync` returned true on [this line](url) because `lastDequeuedMessage` was reset, but now it returns false which leads to getLastMessageIdAsync execution. And because seek drops connection, `getLastMessageIdAsync` fails for me with server error "_Error: MetadataError. Message: Consumer not found_", sot the test fails

Reply:
The logic w/o this PR is problematic because `lastDequeueMessage` is insufficient to express seek, startMessageId and normal dequeue behavior. Therefore lead the bug as this PR aims to fix.

Therefore, If it breaks your test, please reconsider your code logic by using the `seekMessageId` or `lastDequeueMessage`. Since the master branch is in the health state

Reply:
Yes, I like this improvement, just want to add that additional effort needs to be done here to make it work properly, since Seek+Read is a very common reader pattern. If I put a small delay between 2 and 3 step, then everything works properly, because connection `State` changes to `Connecting` and `getLastMessageId` is retried, but if `hasMessageAvailableAsync` is called before state change, then I get the error.
To make it clear - I don't run the code from the branch, but porting changes to .net client library, so this is how I found the issue, but it looks like this issue logically exists for Java client as well.

Reply:
Sorry, fixed the link. No, reader is used by single thread, it waits for seek result and immediately calls hasMessageAvailableAsync

Reply:
There is another issue with the same test with Batching enabled - as long as `lastDequeueMessage` is not reset, it stays on last `BatchMessageIdImpl` while `getLastMessageIdAsync` always returns `MessageIdImpl` so [this check](https://github.com/apache/pulsar/blob/branch-2.5/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L1603) returns false because types of messageIds are different

Reply:
FYI, adding [this line](https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L1542) fixes both issues for me at the moment